File: Medium/3217. Delete Nodes From Linked List Present in Array.java
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        final var set = new boolean[100001];
        for (int num : nums) {
            set[num] = true;
        }
        while (set[head.val]) {
            head = head.next;
        }
        final var res = head;
        while (head.next != null) {
            if (set[head.next.val]) {
                head.next = head.next.next;
            } else {
                head = head.next;
            }
        }
        return res;
    }
}

File: Medium/3217. Delete Nodes From Linked List Present in Array.py
# Definition for singly-linked list.
from typing import List, Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        to_remove = set(nums)
        result = dummy = ListNode(0, head)
        while head:
            if head.val in to_remove:
                dummy.next = head.next
            else:
                dummy = head
            head = head.next
        return result.next